tests/test_timestamp.py

```diff
@@ -0,0 +1,138 @@
+#
+#    http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS
+# OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the
+# License.
+
+# Python 2/3 compatibility
+from decimal import Decimal
+
+import pytest
+
+from amazon.ion.core import Timestamp, TimestampPrecision, record
+from tests import parametrize
+
+
+class _P(record('timestamps', 'expected_microseconds')):
+    def __str__(self):
+        return self.desc
+
+
+MISSING_MICROSECOND = _P(
+    timestamps=[Timestamp(
+            2011, 1, 1,
+            0, 0, 0, None,
+            precision=TimestampPrecision.SECOND, fractional_precision=6, fractional_seconds=0
+            ),
+            Timestamp(
+            2011, 1, 1,
+            0, 0, 0, None,
+            precision=TimestampPrecision.SECOND, fractional_precision=6, fractional_seconds=Decimal('0.123456')
+            ),
+            Timestamp(
+            2011, 1, 1,
+            0, 0, 0, None,
+            precision=TimestampPrecision.SECOND, fractional_precision=6, fractional_seconds=Decimal('0.123456789')
+            ),
+            Timestamp(
+            2011, 1, 1,
+            0, 0, 0, None,
+            precision=TimestampPrecision.SECOND, fractional_precision=6, fractional_seconds=Decimal('0.000123')
+            ),
+            Timestamp(
+            2011, 1, 1,
+            0, 0, 0, None,
+            precision=TimestampPrecision.SECOND, fractional_precision=6, fractional_seconds=Decimal('0.123000')
+            )],
+    expected_microseconds=[0, 123456, 123456, 123, 123000]
+)
+
+
+@parametrize(
+    MISSING_MICROSECOND
+)
+def test_missing_microsecond(item):
+    for i in range(len(item.timestamps)):
+        assert item.timestamps[i].microsecond == item.expected_microseconds[i]
+
+
+class _P(record('timestamps', 'expected_fractional_precision')):
+    def __str__(self):
+        return self.desc
+
+
+MISSING_FRACTIONAL_PRECISION = _P(
+    timestamps=[Timestamp(
+            2011, 1, 1,
+            0, 0, 0, 0,
+            precision=TimestampPrecision.SECOND, fractional_precision=6, fractional_seconds=None
+            ),
+            Timestamp(
+            2011, 1, 1,
+            0, 0, 0, 123456,
+            precision=TimestampPrecision.SECOND, fractional_precision=6, fractional_seconds=None
+            ),
+            Timestamp(
+            2011, 1, 1,
+            0, 0, 0, 123,
+            precision=TimestampPrecision.SECOND, fractional_precision=6, fractional_seconds=None
+            ),
+            Timestamp(
+            2011, 1, 1,
+            0, 0, 0, 123000,
+            precision=TimestampPrecision.SECOND, fractional_precision=6, fractional_seconds=None
+            )],
+    expected_fractional_precision=[0, Decimal('0.123456'), Decimal('0.000123'), Decimal('0.123000')]
```

Comment:
Ditto regarding the record.

I think you are misunderstanding what we're doing with the *parameter* data structures, they are to represent an encapsulation of a test case as data.  So instead of having one *parameter* with two lists, you should have a tuple of *parameters* that you feed in to the test case where *each* parameter is the description of the test case.  In your case the `Timestamp` you constructed and some fields you're predicating on.

Reply:
Done.

Reply:
I have follow up below, but resolving this instance.